tests/testthat/test-a-recording.R

```diff
@@ -43,3 +43,62 @@ test_that("We can specify the path when starting capture", {
   dbDisconnect(con)
   stop_db_capturing()
 })
+
+test_that("dbGetQuery error checking", {
+  # Check to make sure con was not created before start_db_capturing()
+  suppressMessages(con <- nycflights13_create_sqlite(verbose = FALSE))
+
+  start_db_capturing()
+
+  # Testthat sets this to "tests/testthat//_memory_"
+  # Setting this to NULL so it will mimic a developers experience
+  .dittodb_env$db_path <- NULL
+
+  error_get_query <- expect_error(
+    object = dbGetQuery(con, "SELECT * FROM airlines"),
+    regexp = "^Database capture failed"
+  )
+  expect_equal(error_get_query$n, 1)
+
+  # The following functions trigger two errors
+  # testthat_transition( )
+  regex_empty <- "^$"
+
+  testthat_transition(
+    old = expect_error(dbListTables(con)),
+    new = expect_error(expect_error(dbListTables(con)), regex_empty)
+  )
+
+  testthat_transition(
+    old = expect_error(dbListFields(con, "airlines")),
+    new = expect_error(expect_error(dbListFields(con, "airlines")), regex_empty)
+  )
+
+  testthat_transition(
+    old = expect_error(dbExistsTable(con, "airlines")),
+    new = expect_error(expect_error(dbExistsTable(con, "airlines")), regex_empty)
+  )
+
+  error_tbl <- expect_error(dplyr::tbl(con, "airlines"))
+  expect_null(error_tbl$n)
+
+  suppressWarnings(dbDisconnect(con))
+  stop_db_capturing()
+})
+
+# test_that("dbListTables error checking ", {
+#   # Check to make sure con was not created before start_db_capturing()
+#   suppressMessages(con <- nycflights13_create_sqlite(verbose = FALSE))
+#
+#   start_db_capturing()
+#
+#   # Testthat sets this to "tests/testthat//_memory_"
+#   # Setting this to NULL so it will mimic a developers experience
+#   .dittodb_env$db_path <- NULL
+#
+#   x <- expect_error(expect_error(dbListTables(con)))
+#   expect_equal(x$n, 2)
+#
+#   suppressWarnings(dbDisconnect(con))
+#   stop_db_capturing()
+# })
```

Comment:
Is this needed? It looks like it's duplicated — if it is let's remove it

Reply:
fixed